system/jlib/jbuff.hpp

```diff
@@ -325,7 +325,7 @@ class jlib_decl Variable2IDataVal : implements IDataVal
 
     virtual const void * data() const { return *pData; };
     virtual void clear() { free(*pData); *pData = NULL; *pLen = 0; };
-    virtual void setLen(const void * val, size_t length) { free(*pData); *pData = malloc(length); memcpy(*pData, val, length); *pLen = (size32_t)length; }
+    virtual void setLen(const void * val, size_t length) { free(*pData); *pData = malloc(length); memcpy_iflen(*pData, val, length); *pLen = (size32_t)length; }
```

Comment:
Should we move this earlier, ie if length == 0 then why malloc(0) ? altho probably ok on most systems.

Reply:
It could, but then you would need conditional code for clearing pData and plen.  in this case I went for the smallest change.